internal/http/services/owncloud/ocdav/trashbin.go

```diff
@@ -469,6 +469,26 @@ func (h *TrashbinHandler) restore(w http.ResponseWriter, r *http.Request, s *svc
 		return
 	}
 
+	// Restoring to a non-existent location is not supported by the WebDAV spec. The following block ensures the target
+	// restore location exists, and if it doesn't returns a conflict error code.
+	if dstStatRes.Status.Code == rpc.Code_CODE_NOT_FOUND && strings.HasSuffix(dst, itemPath) && isNested(dst) {
+		parentStatReq := &provider.StatRequest{
+			Ref: &provider.Reference{Path: path.Join(getHomeRes.Path, filepath.Dir(dst))},
+		}
+
+		parentStatResponse, err := client.Stat(ctx, parentStatReq)
+		if err != nil {
+			sublog.Error().Err(err).Msg("error sending grpc stat request")
+			w.WriteHeader(http.StatusInternalServerError)
+			return
+		}
+
+		if parentStatResponse.Status.Code == rpc.Code_CODE_NOT_FOUND {
+			HandleErrorStatus(&sublog, w, &rpc.Status{Code: rpc.Code_CODE_ALREADY_EXISTS})
```

Comment:
I'm not sure I follow this. If the destination's parent directory doesn't exist, why are we reporting already exists?

Reply:
we can go with
> (c) Use `FAILED_PRECONDITION` if the client should not retry until the system state has been explicitly fixed. E.g., if an "rmdir" fails because the directory is non-empty, `FAILED_PRECONDITION` should be returned since the client should not retry unless the files are deleted from the directory.